app/src/main/java/com/github/studydistractor/sdp/account/FirebaseUser.kt

```diff
@@ -0,0 +1,51 @@
+package com.github.studydistractor.sdp.account
+
+import android.util.Log
+import androidx.compose.runtime.mutableStateListOf
+import androidx.compose.runtime.snapshots.SnapshotStateList
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.database.DataSnapshot
+import com.google.firebase.database.DatabaseError
+import com.google.firebase.database.FirebaseDatabase
+import com.google.firebase.database.ValueEventListener
+
+class FirebaseUser : UserInterface {
+    private val auth = FirebaseAuth.getInstance()
+    private val db = FirebaseDatabase.getInstance()
+
+    private val FRIENDSPATH = "Friends"
+
+
+    override fun getCurrentUid(): String? {
+        return auth.uid
+    }
+
+    override fun addNewFriend(uid1: String, uid2: String) {
+        db.getReference(FRIENDSPATH).child(uid1).child(uid2).setValue(uid2)
```

Comment:
Having some feeback is nice, see this [PR](https://github.com/StudyDistractor/sdp/pull/101) by Athena to see how she did it. And don't forget to remove the listener when it is not needed anymore.